services/QuillLMS/spec/services/classlink_integration/user_persister_spec.rb

```diff
@@ -48,6 +48,17 @@
       end
     end
 
+    context 'when non classlink user already exists' do
+      let(:existing_provider_users) { [] }
+      let!(:existing_non_classlink_user) { create(:user, role: 'student', email: incoming_user[:email]) }
+
+      it 'creates new Account record' do
+        expect { subject.run }
+          .to change(ClasslinkAccount, :count).by(1)
+          .and not_change(User, :count)
+      end
+    end
```

Comment:
Again, is it worth writing a test case for update behavior when a match is found by email, but (for instance) name in Classlink does not match name in LMS database?

Reply:
Yes, here it makes sense to express that an incoming new name should be updated. I've added that to the spec.